A3A/addons/core/functions/Missions/fn_LOG_Salvage.sqf

```diff
@@ -84,6 +85,45 @@ private _typeVeh = if (_difficultX) then { selectRandom (_faction get "vehiclesG
 private _typeGroup = if _difficultX then {selectRandom ([_faction, "groupsTierSquads"] call SCRT_fnc_unit_flattenTier)} else {selectRandom ([_faction, "groupsTierMedium"] call SCRT_fnc_unit_flattenTier)};
 private _boatSpawnLocation = selectRandom [_mrk1Pos, _mrk2Pos, _mrk3Pos];
 
+if ("vanilla" in A3A_factionEquipFlags) then {
+	private _typeSDV = "";
+	private _diverType = "";
+	private _diversGroup = createGroup _sideX;
+	private _diversGroup2 = createGroup _sideX;
+   _typeSDV = selectRandom (_faction get "vehiclesSDV");
+   if (_typeSDV == "I_SDV_01_F") then {
+		_diverType = "I_diver_F";
+   };
+   if (_typeSDV == "O_SDV_01_F") then {
+		_diverType = "O_diver_F";
+   };
+   if (_typeSDV == "B_SDV_01_F") then {
+		_diverType = "B_diver_F";
+   };
+   for "_i" from 1 to 2 do {
+		private _SDVmarker = selectRandom [_mrk1Pos, _mrk2Pos, _mrk3Pos];
+		private _vehSDV = createVehicle [_typeSDV, _SDVmarker, [], 30, "NONE"];
+		_vehSDV setPos [getPos _vehSDV select 0, getPos _vehSDV select 1, (getPos _vehSDV select 2) - 7];
+		[_vehSDV, _sideX] call A3A_fnc_AIVEHinit;
+		for "_i" from 1 to 2 do {
+			private _diver = _diversGroup createUnit [_diverType, getPos _vehSDV, [], 10, "NONE"];
+			_diver setPos [getPos _diver select 0, getPos _diver select 1, (getPos _diver select 2) - 8];
+			[_diver] call A3A_fnc_NATOinit;
+   		};
+		for "_i" from 1 to 2 do {
+			private _diver = _diversGroup2 createUnit [_diverType, getPos _vehSDV, [], 10, "NONE"];
+			_diver setPos [getPos _diver select 0, getPos _diver select 1, (getPos _diver select 2) - 8];
+			[_diver] call A3A_fnc_NATOinit;
+   		};
+   };
+	private _diversWaypoint = _diversGroup addWaypoint [selectRandom _markers, 30];
+	_diversWaypoint setWaypointType "SAD";
+	_diversGroup setCurrentWaypoint _diversWaypoint;
+	private _divers2Waypoint = _diversGroup2 addWaypoint [selectRandom _markers, 30];
+	_divers2Waypoint setWaypointType "SAD";
+	_diversGroup2 setCurrentWaypoint _divers2Waypoint;
+};
```

Comment:
We shouldn't necessarily assume that templates with the "vanilla" flag also have vehiclesSDV defined